Update balenaFin to Buster

alexgg opened the pull request.

Commit:
Add initial balenafin support

Signed-off-by: Andrei Gherzan <andrei@balena.io>

XECDesign committed:
Handle cases where binfmt_misc is built into the kernel

Fixes #264

XECDesign committed:
Update Dockerfile

XECDesign committed:
Replace pxz with xz -T0

lurch committed:
Switch to using parted's machine parseable output (#270)

XECDesign committed:
stage2: ethtool

XECDesign committed:
stage2: install rng-tools

https://github.com/raspberrypi/linux/issues/2799

XECDesign committed:
Update release notes

XECDesign committed:
Add new changelog entry

hhromic committed:
Silence shellcheck warnings (#279)

* SC1091: the `config` file might not be present, which is normal.
* SC2086: Double quote to prevent globbing and word splitting.

Tested clean output using: `find -name "*.sh" -exec shellcheck -x {} \;`

hhromic committed:
Do not assume the changelog file is always present for the info file (#280)

The `(..)/raspberrypi-kernel/changelog.Debian.gz` file is not guaranteed
to be present in the built `ROOTFS_DIR`, for example when building very
minimal images without package documentation. In these cases, the `firmware`
variable will be left empty and the subsequent calls to `curl` will return
large 404 HTML content from GitHub.

Instead, simply check if the changelog file exists before using it.

XECDesign committed:
export-image: Remove interfaces.dpkg-old

samtygier committed:
Better docs for STAGE_LIST (#289)

Note that quotes are needed, see bug #288.

XECDesign committed:
stage2: install pciutils

XECDesign committed:
cmdline.txt: remove dwc_otg.lpm_enable=0

XECDesign committed:
build.sh: run quilt upgrade before editing patches

XECDesign committed:
Update to Buster

Remove ui and staging components

XECDesign committed:
Remove Oracle Java

XECDesign committed:
export-image: Set /boot size to 256MB

XECDesign committed:
os_list: add checksums for PINN

XECDesign committed:
os_list: add Pi 4 support

XECDesign committed:
os_list: Bump kernel to 4.19

XECDesign committed:
config.txt: Enable FKMS and set max_framebuffers=2 in [pi4]

XECDesign committed:
stage4: remove timidity

XECDesign committed:
partition_setup.sh: Fix cmdline.txt handling

Add sdhci.debug_quirks2=4 to cmdline.txt when resizing

Only remove 'quiet' from cmdline if splash screen is off

XECDesign committed:
stage2: Disable wifi on 5GHz models

XECDesign committed:
script/common: Handle aarch64 host builds

XECDesign committed:
stage4: enable xcompmgr

XECDesign committed:
Java 8 workaround

Java 8 segfaults during install.
Installing OpenJDK 11 first prevents Java 8 from running.

XECDesign committed:
Replace IDLE with Thonny in stage 4 and add Mu to stage 5

XECDesign committed:
stage3: include arandr

XECDesign committed:
Fix Buster packages

XECDesign committed:
stage5: Remove wolfram-engine until Buster version is available

XECDesign committed:
stage4: fix autologin

XECDesign committed:
export-image: remove /etc/network/interfaces.dpkg-old

XECDesign committed:
Disable ld.so.preload while building

XECDesign committed:
export-noobs: update OS.png

XECDesign committed:
stage5: install greenfoot-unbundled

XECDesign committed:
Revert "Java 8 workaround"

This reverts commit 3682a747bdf319a34fcce7cad9771a1a3ead34ec.

XECDesign committed:
Update release notes

hhromic committed:
Update Dockerfile base image to Debian Buster (#295)

With the Qemu version shipped with Debian Stretch, the `man-db` package
being installed for Buster in the image triggers many of these errors:

    qemu: Unsupported syscall: 383

This is a manifestation of the following bug:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=891109

This is resolved in the current Qemu version shipped with Debian Buster.

ryanteck committed:
64 bit message (#298)

Added a handy message to make it clear that compiling on a 64 bit OS is causing issues. After spending over an hour of troubleshooting think it should be made clearer as it'll affect quite a few users.

rkubes committed:
Fix build.sh to run dependencies_check prior to using curl. (#299)

oskar456 committed:
Remove obsolete IPv6 turnoff (#300)

Turning off IPv6 by aliasing `net-pf-10` to `off` does not work anymore.
Also, turning off IPv6 on system level breaks apps depending on IPv6
loopback or IPv6 link-local addresssing and creates issues in both
dual-stack and IPv6-only environments.

Signed-off-by: Ondřej Caletka <ondrej@caletka.cz>

xranby committed:
stage2/00-copies-and-fills/02-run.sh: Fix re-runs of script (#301)

Check that file exist before move.

macisamuele committed:
build-docker.sh does not assume PWD=<repo root> (#302)

* bash variables in build-docker.sh are wrapped by curly brackets

* Ensure presence of config file while running build-docker.sh

* Do not assume that build-docker.sh is run from the repository root directory

* Mount config file in predictable location in docker container

gszy committed:
README.md shouldn’t be executable (#303)

In 667318116acda97b58734ae7512789a0f56db306, README.md’s file mode bits
were changed from 0o644 to 0o755. This commit reverts them back to
0o644.

gszy committed:
Revert "stage2: Add apt-transport-https" (#304)

This reverts commit 1806504983024db6cf00935e20abcfd3295ab248.

In Buster, APT has built‐in support for HTTPS repos (since version 1.5).
The ca-certificates package is already included in the modified file, so
this commit shouldn’t break anything.

XECDesign committed:
stage5: Add Mathematica

XECDesign committed:
stage2: Add vl805fw

XECDesign committed:
Update release notes

XECDesign committed:
Add missing release notes

XECDesign committed:
stage3: remove Epiphany

XECDesign committed:
stage2: add ntfs-3g

132ikl committed:
Update Stretch to Buster in README (#310)

hilli committed:
Make build_docker.sh portable again (#308)

Last commit made the script break on macOS.

From `man sed` (On Linux):

```
       -E, -r, --regexp-extended

              use extended regular expressions in the script (for portability use POSIX -E).
```

hhromic committed:
Ensure that the configuration file is an absolute path in Docker build (#306)

* Use `&&` instead of `;` in Docker pipeline

* In case of error, `&&` does not continue execution

* Silence shellcheck warning

* SC2086: Double quote to prevent globbing and word splitting.

* Ensure that the configuration file is an absolute path in Docker build

The specific problem is in commit 2ddd7c1, where the passed config file
(using the `-c` option) is now mounted inside the container using the
`--volume src:dest:opt` Docker option.

The problem is that Docker requires absolute paths for mounting single
files inside the container, otherwise it silently tries to mount a volume
name instead as an empty directory. Therefore the Docker build no longer
works with the following invocation forms (relative config-paths):

    ./build-docker.sh -c myconfig
    /path/to/build-docker.sh -c myconfig   # also doesn't work

This commit uses `realpath` (included in coreutils) in the Docker build
script to ensure that the passed configuration file is always an
absolute path before passing it to Docker.

kevinastone committed:
Moved git-hash determination to passed in build variable (#273)

christophehurpeau committed:
Remove unused interfaces file (#322)

XECDesign committed:
stage5: add scratch3

XECDesign committed:
stage4: rename thonny package

jbussdieker committed:
Support regional config settings (#323)

XECDesign committed:
Revert "export-image: Remove interfaces.dpkg-old"

This reverts commit a016561600292047971b65d034f0e6eaeadd1122.

Commit:
Set IMG_NAME in build script

Change-type: minor
Signed-off-by: Giovanni Garufi <giovanni@balena.io>

rkubes committed:
Updated export-image to not depend on fake-hwclock and hardlink. (#326)

XECDesign committed:
Include rpi-eeprom

XECDesign committed:
Update release notes

fpicalausa committed:
Update config.txt to replace lirc-rpi with gpio-ir (#328)

The example for enabling IR transmission in `/boot/config.txt` is still using the deprecated `lirc-rpi` overlay. 
The documentation in `/boot/overlays/README` indicates that this overlay has been deprecated in favor of `gpio-ir` / `gpio-ir-tx`.

This updates the actual config.txt to suggest `gpio-ir` instead of `lirc-rpi`.

XECDesign committed:
stage3: Mousepad used as simple text editor instead of leafpad

matthijskooijman committed:
Use parted for partitioning (#285)

Previously, fdisk was used by sending commands into its stdin, which is
not very robust (since it heavily relies on the interactive prompts
offered by fdisk as well as the default values it offers, which seem
prone to changing in future version).

It seems likely that in the past, fdisk was easier than parted since it
provides default values that make it easier to create adjacent
partitions, without precalculating all positions in the script. However
now that partitions are manually being aligned, all data must be
calculated anyway.

This commit changes the partition generation to use parted rather than
fdisk. For this, it rewrites various calculations and renames variables
to be easier to read as well. All values are now in number of bytes,
rather than mixing bytes and sectors.

This commit also makes makes sure that the boot partition and root
partition are always adjacent (previously the root partition was aligned
without also rounding the boot partition size, leaving some empty space
in between).

As a side effect of using parted, this also causes the "bootcode" part
of the MBR to be filled with some default x86 bootcode. This is totally
irrelevant for booting the Raspberry Pi, but it does prevent triggering
a bug in parted. When using parted to change the partition table (e.g.
when resizing the root partition on first boot by raspi-config's
init_resize.sh), the disk identifier would be changed due to this bug,
which would change the PARTUUID of all partitions. The init_resize.sh
script would work around this by updating the PARTUUID in e.g. fstab,
but that's fragile at best.  This commit prevents the bug from
triggering and keeps the disk identifier the same.

See https://debbugs.gnu.org/35714 for details about this parted bug.

This commit fixes #284.

XECDesign committed:
Update release notes

XECDesign committed:
Update release notes

matthijskooijman committed:
Document how to get values for the localization values

XECDesign committed:
Update changelog

reneklootwijk committed:
Added hostname parameter

XECDesign committed:
Ensure debootstrap runs through 'sh' rather than 'bash'

This fixes #338 and fixes #294

XECDesign committed:
Add hostname to /etc/hosts

snoe925 committed:
OS X does not have realpath (#342)

Check that realpath is present.  OS X doesn't have realpath.  The RPi builds on Docker for Mac if you increase the VM size. I am using 256G for my image.  See the Docker for Mac preferences.

LCvanDinteren committed:
Export base directory before loading config file (#347)

LCvanDinteren committed:
Remove accidental trailing brace (#348)

LCvanDinteren committed:
Explicitly check if postrun script exists in BASE_DIR (#351)

Commit:
New types of images. (#352)

Commit:
Added stage 5 to the text (#353)

chrisdecker1201 committed:
configure unprotected wifi when WPA_PASSWORD is unset (#357)

XECDesign committed:
Rename HOSTNAME variable to TARGET_HOSTNAME

Bash sets the HOSTNAME variable, overriding our default

Commit:
Updated info about stage4 and stage5 (#359)

chrisdecker1201 committed:
move picraft to stage where minecraft is installed (#362)

samtygier committed:
Use tee with wpa_passphrase so errors are visable

wpa_passphrase writes its error messages to stdout, so this needs to
tee so that errors are visable to user. Also need to enable pipefail
so that the error code still gets caught.

samtygier committed:
Check WPA_PASSWORD length early

WPA_PASSWORD needs to be between 8 adn 63 characters. Check early
to avoid hitting error in stage2.

samtygier committed:
WPA_PASSWORD length check should only occur if set

Test if WPA_PASSWORD is set before check that it has a valid length.

Commit:
README.md - mention WPA_PASSWORD length requirement

Commit:
capitalise Raspberry Pi

XECDesign committed:
Update ROOT_MARGIN

XECDesign committed:
stage5: install code-the-classics

XECDesign committed:
Update MagPi URL scheme

XECDesign committed:
Update release notes

XECDesign committed:
Update persistent rkfill files

XECDesign committed:
Update release notes

XECDesign committed:
Update release notes

XECDesign committed:
Update release notes

XECDesign committed:
Update release notes

XECDesign committed:
Update release notes

luigifcruz committed:
Retry apt downloads 3 times

Fixes #194

timvisee committed:
BusyBox realpath does not support -s flag, fallback to use no flags (#384)

* BusyBox realpath does not support -s flag, fallback to use no flags

* Fix config sourcing problem

lpenz committed:
Dockerfile: use --no-install-recommends in apt-get install (#383)

DragonForgedTheArtist committed:
Exposed RELEASE as a configurable variable (#391)

Exposed RELEASE as a configurable variable and updated README.md.  As a
bonus, there is only on place to change the release name when the next
release is stable now.

XECDesign committed:
Bookshelf PDF download

XECDesign committed:
Update release notes

XECDesign committed:
Add new changelog entry

XECDesign committed:
stage2: install raspinfo

XECDesign committed:
Update release notes

XECDesign committed:
Raspbian -> Raspberry Pi OS

XECDesign committed:
Update OS.png

Commit:
Enable adding SSH pubkey and setting pubkey only (#380)

This commit add the ability to specify an SSH public key as well as the
option to disable password authentication and only allow public key
authentication for SSH.

Signed-off-by: Daniel F. Dickinson <cshored@thecshore.com>

Commit:
Update README.md

rnestler committed:
Only disable wifi if WPA_COUNTRY is not set (#416)

* Only disable wifi if WPA_COUNTRY is not set

* Use '-n' instead of '-v' to test if WPA_COUNTRY is set

Apparently -v doesn't work in all shells.

Commit:
workaround build failure on AWS M6g instances (#449)

aniongithub committed:
Automagically use i386/debian:buster (#415)

* Autmagically use 1386/debian:buster when running on 64-bit host to prevent error #271

XECDesign committed:
stage4: add printing support

XECDesign committed:
export-image: get apt lists from public repo

alexgg committed:
balenafin: Update to buster

Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
balenafin: Use sd8887-nxp instead of sd8887-mrvl

NXP has released this wireless driver under a GPL license so let's use
it instead of having a module-only package.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
balenafin: Move to using cloudsmith as apt repository

bintray has ceased to provide service.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg committed:
build-docker: Check for node before attempting mknod

If the block devices already exists do not attempt to create them or
the script will exit in error.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg commented:
@balena-ci test

alexgg commented:
@balena-ci retest

alexgg commented:
@balena-ci retest

alexgg commented:
@balena-ci help

ghost commented:
Here's a list of commands supported by VersionBot
- `@balena-ci I self-certify!` - use [the self-certification flow](https://docs.google.com/drawings/d/1aoXCdi8bzinxaLM0BoKMJaiZjIJj5Az02Npw_L2iQkE/edit?usp=sharing) for this PR
- `@balena-ci test` - trigger the tests for this PR
- `@balena-ci version` - generate the versioned branch for your PR
- `@balena-ci rebase` - rebase the PR

alexgg commented:
@balena-ci test

alexgg committed:
build-docker: Set default path for configuration file

If left undefined the script ends when run with -e with unbound
variable.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg commented:
This is waiting feedback on the testing required.
https://www.flowdock.com/app/rulemotion/r-hardware/threads/wDIy3kKGU_URz8B-5dlkKAPztJP

alexgg committed:
build-docker: Replace binfmt mount with QEMU interpreters registration

Instead of just mounting the miscellaneous binary format kernel support,
make sure that the QEMU interpreters are also installed.

Running a privileged docker container avoids having to sudo the mount
command.

Change-type: patch
Signed-off-by: Alex Gonzalez <alexg@balena.io>

alexgg commented:
Won't do
